### PR TITLE
71 feature 291 récupération personnes disponible pour conversation

### PR DESCRIPTION
**Pull request by quentinmananes**

### CODE REVIEW

File: src/__tests__/routes/adm/class/register.test.js
Comment (quentinmananes) [inline src/__tests__/routes/adm/class/register.test.js:1]: Change from 2.4.2 (needed)

File: src/config/db.default.js
Comment (quentinmananes) [inline src/config/db.default.js:1]: Change from 2.4.2 (needed)

File: src/models/classes.js
Comment (quentinmananes) [inline src/models/classes.js:1]: Change from 2.4.2 (needed)

File: src/models/users.js
Comment (quentinmananes) [inline src/models/users.js:1]: Change from 2.4.2 (needed)

File: src/package.json
Comment (quentinmananes) [inline src/package.json:1]: Change from 2.4.2 (needed)

File: src/routes/adm/class/register.js
Comment (quentinmananes) [inline src/routes/adm/class/register.js:1]: Change from 2.4.2 (needed)

File: src/routes/adm/csvRegisterUser.js
Comment (quentinmananes) [inline src/routes/adm/csvRegisterUser.js:1]: Change from 2.4.2 (needed)

File: src/routes/adm/register.js
Comment (quentinmananes) [inline src/routes/adm/register.js:1]: Change from 2.4.2 (needed)

File: src/routes/adm/router.js
Comment (quentinmananes) [inline src/routes/adm/router.js:1]: Change from 2.4.2 (needed)

File: src/routes/user/chat/getAvailableChatUsers.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const agg = [
+      {
+        $match: {
+          facility: req.user.facility,
+          _id: { $nin: [req.user._id] },
+          role: { $in: roles }
+        }
+      }
+    ]
+    const result = await Users.aggregate(agg)
```
Comment (Exiels) [inline src/routes/user/chat/getAvailableChatUsers.js:39]: Beware of what you get from the aggregation. Do not send things like password, firstConnexion, facility and email.

Lookup and unwind classes and role

File: src/swagger.yaml [hunk 2]
```diff
@@ -221,6 +221,26 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ArrayProfiles'
+        '500':
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/internalServerError'
```
Comment (Exiels) [inline src/swagger.yaml:243]: Change the response to correspond to what the modified aggregation send

**Review by Exiels — CHANGES_REQUESTED** [review]

**Review by Exiels — APPROVED** [review]

**Pull request merged**